Dp 1797 update athena_load to use existing schema for curated table

tom-webber opened the pull request:
resolves #1797:

* `athena_load` process uses type and comment/description (trimmed where necessary) data from the registered schema to create the curated parquet files. Be confident metadata registered in the glue catalog matches that in the s3 schema.
* data product database names are suffixed with their major version suffix.
* create tests for any additional functions required.

Commit:
Use existing schema for curated table if exists
test load handler with and without existing schema

Commit:
refactor athena_load_handler tests
fix bare except

Commit:
database name for DataProductElement.curated_data_table includes latest major version
trim schema table desc and col comment to glue limits

Commit:
clear defaults from glue schema template
clarify function definition

Commit:
linting

Commit:
update changelogs and upversion

tom-webber committed:
Merge branch 'main' into DP-1797-update-athena-load-to-use-existing-schema-for-curated

tom-webber commented:
Note that the truncating of the glue schema column comments and table description are currently **lossy** for large descriptions as these functions are manipulating the data that is saved. Future work will save a non-glue version of the metadata, so this implementation here will not be lossy in that case.

tom-webber committed:
Merge branch 'main' into DP-1797-update-athena-load-to-use-existing-schema-for-curated

LavMatt commented on containers/daap-athena-load/src/var/task/athena_load_handler.py:
```diff
@@ -26,12 +27,15 @@ def handler(
     extraction = RawDataExtraction.parse_from_uri(full_s3_path)
     data_product_element = extraction.element
 
+    data_product_name = data_product_element.curated_data_table.database
```
We now need to distinguish between the data_product_name and the database name that has a version suffix, otherwise you'll get a schema doesn;t exist error, e.g. looking for `{data_product_name}_v1`
```suggestion
    data_product_name = data_product_element.data_product.name
    database_name = data_product_element.curated_data_table.database

LavMatt commented on containers/daap-athena-load/src/var/task/athena_load_handler.py:
```diff
@@ -41,6 +45,23 @@ def handler(
     )
 
+    try:
+        table_schema = (
```
what i've suggested does mean we might need to add another attribute to `DataProductSchema` or at least have it populate DatabaseName with the data_product_name+version suffix

could make sure that key's value in the schema includes the suffix in `DataProductSchema.convert_schema_to_glue_table_input_csv()`

tom-webber replied:
Good point.

I've added `database_name` to the `BaseJsonSchema` as it can be calculated there. It's then inherited into `DataProductMetadata` and `DataProductSchema` classes. 

This is used in the convert method for `DatabaseName`

LavMatt reviewed (commented):
Couple of minor suggestions re. the suffix as it will fail to find the schema as is. We don't save the data product with a version suffix, that's only for database name

tom-webber committed:
Update containers/daap-athena-load/src/var/task/athena_load_handler.py

Co-authored-by: Matt <38562764+LavMatt@users.noreply.github.com>

Commit:
add `database_name` to `BaseJsonSchema` - used in glue schema
add `database_name` to `DataProductElement` - used in `curated_data_table`
update tests

tom-webber committed:
Merge branch 'main' into DP-1797-update-athena-load-to-use-existing-schema-for-curated

Commit:
update changelogs

Commit:
linting

tom-webber committed:
Merge branch 'main' into DP-1797-update-athena-load-to-use-existing-schema-for-curated

LavMatt reviewed (approved):
Looks good now
